tests/test.sh

```diff
@@ -854,6 +854,74 @@ zero_local_jobs_test()
     echo
 }
 
+environment_extraction_priority_test()
+{
+    echo Running environment_extraction_priority_test test.
+    reset_logs remote "$@"
+
+    mkdir "$testdir/new_test/" &> /dev/null
+
+    test_priorities=(10 12 15)
+    for compile_priority in ${test_priorities[@]}; do
+        kill_daemon remoteice1
+        start_iceccd remoteice1 -p 10246 -m 2
```

Comment:
We have to restart the daemon here to clear it's cache which contains the extracted environment from a previous build. If this is not done we won't see tar executing.